packages/cli/src/services/cache.service.ts

```diff
@@ -0,0 +1,167 @@
+					? `(Cache Service) Initialising Redis cluster connection with nodes: ${clusterNodes
+							.map((e) => `${e.host}:${e.port}`)
+							.join(',')}`
+					: `(Cache Service) Initialising Redis client connection with host: ${
+							host ?? 'localhost'
+					  } and port: ${port ?? '6379'}`,
+			);
+			const sharedRedisOptions: RedisOptions = {
+				username,
+				password,
+				db,
+				enableReadyCheck: false,
+				maxRetriesPerRequest: null,
+			};
+			const redisClient = usesRedisCluster
+				? new Redis.Cluster(
+						clusterNodes.map((node) => ({ host: node.host, port: node.port })),
+						{
+							redisOptions: sharedRedisOptions,
+						},
+				  )
+				: new Redis({
+						host,
+						port,
+						...sharedRedisOptions,
+						retryStrategy: (): number | null => {
+							const now = Date.now();
+							if (now - lastTimer > 30000) {
+								// Means we had no timeout at all or last timeout was temporary and we recovered
+								lastTimer = now;
+								cumulativeTimeout = 0;
+							} else {
+								cumulativeTimeout += now - lastTimer;
+								lastTimer = now;
+								if (cumulativeTimeout > redisConnectionTimeoutLimit) {
+									LoggerProxy.error(
+										`Unable to connect to Redis after ${redisConnectionTimeoutLimit}. Exiting process.`,
+									);
+									process.exit(1);
+								}
+							}
+							return 500;
+						},
+				  });
+			// #endregion TEMPORARY Redis Client Code
+			const redisStore = redisInsStore(redisClient, {
+				ttl: config.getEnv('cache.redis.ttl'),
+			});
+			this.cache = await caching(redisStore);
+		} else {
+			this.cache = await caching('memory', {
+				ttl: config.getEnv('cache.memory.ttl') ?? 10 * 1000,
+				max: config.getEnv('cache.memory.max') ?? 100 * 1024 * 1024,
+			});
+		}
+	}
+
+	async uninit() {
+		if (this.cache) {
+			await this.reset();
+			this.cache = undefined;
+		}
+	}
+
+	getCacheInstance(): RedisCache | MemoryCache | undefined {
+		return this.cache;
+	}
+
+	async get<T>(key: string): Promise<T> {
+		if (!this.cache) {
+			await this.init();
+		}
+		return this.cache?.store.get(key) as T;
+	}
+
+	async set<T>(key: string, value: T, ttl?: number): Promise<void> {
+		if (!this.cache) {
+			await this.init();
+		}
+		return this.cache?.store.set(key, value, ttl);
+	}
+
+	async del(key: string): Promise<void> {
+		if (!this.cache) {
+			await this.init();
+		}
+		return this.cache?.store.del(key);
+	}
+
+	async reset(): Promise<void> {
+		if (!this.cache) {
+			await this.init();
+		}
+		return this.cache?.store.reset();
+	}
+
+	async keys(): Promise<string[]> {
+		if (!this.cache) {
+			await this.init();
+		}
+		return this.cache?.store.keys() ?? [];
+	}
+
+	async mset<T>(values: Array<[string, T]>, ttl?: number): Promise<void> {
+		if (!this.cache) {
+			await this.init();
+		}
+		return this.cache?.store.mset(values, ttl);
+	}
+
+	async mget<T>(keys: string[]): Promise<Array<[string, T]>> {
+		if (!this.cache) {
+			await this.init();
+		}
```

Comment:
We could add `if (this.cache) return;` to `init()` to avoid checking in every method.

Reply:
the point is that you can use any of the functions immediately from anywhere in the code without having to worry about whether the service is initialized

Reply:
Exactly, I meant that we keep `this.init()` but include the `this.cache` check inside it, and avoid having this negative check in every method, i.e. same end result with less verbiage. But minor thing, feel free to ignore!